sdk/python/kfp_tekton/compiler/_data_passing_rewriter.py

```diff
@@ -343,54 +291,55 @@ def mark_upstream_ios_of_output(template_output, marked_inputs,
         for output in outputs_consumed_as_artifacts
     }
     # task_workspaces = set()
-    for pipeline in pipeline_templates:
-        # Converting pipeline inputs
-        pipeline, pipeline_workspaces = big_data_passing_pipeline(
-            pipeline, inputs_consumed_as_artifacts,
-            output_tasks_consumed_as_artifacts)
+
+    # Converting pipeline inputs
+    pipeline, pipeline_workspaces = big_data_passing_pipeline(
+        pipelinerun_name, pipeline_template, inputs_consumed_as_artifacts,
+        output_tasks_consumed_as_artifacts)
 
     # Add workspaces to pipelinerun if big data passing
-    # Check whether pipelinerun was generated, through error if not.
+    # Check whether pipelinerun was generated, throw error if not.
     if pipeline_workspaces:
-        if not pipelinerun_templates:
-            raise AssertionError(
-                'Found big data passing, please enable generate_pipelinerun for your complier'
-            )
-        for pipelinerun in pipelinerun_templates:
-            pipeline, pipelinerun_workspaces = big_data_passing_pipelinerun(
-                pipelinerun, pipeline_workspaces)
+        pipeline, pipelinerun_workspaces = big_data_passing_pipelinerun(
+            pipelinerun_name, pipelinerun_template, pipeline_workspaces)
 
     # Use workspaces to tasks if big data passing instead of 'results', 'copy-inputs'
     for task_template in container_templates:
         task_template = big_data_passing_tasks(task_template,
                                                inputs_consumed_as_artifacts,
                                                outputs_consumed_as_artifacts)
 
-    # Create pvc for pipelinerun if big data passing.
-    # As we used workspaces in tekton pipelines which depends on it.
-    # User need to create PV manually, or enable dynamic volume provisioning, refer to the link of:
-    # https://kubernetes.io/docs/concepts/storage/dynamic-provisioning
-    # TODO: Remove PVC if Tekton version > = 0.12, use 'volumeClaimTemplate' instead
-    if pipelinerun_workspaces:
-        for pipelinerun in pipelinerun_workspaces:
-            workflow.append(create_pvc(pipelinerun))
+    # # Create pvc for pipelinerun if big data passing.
+    # # As we used workspaces in tekton pipelines which depends on it.
+    # # User need to create PV manually, or enable dynamic volume provisioning, refer to the link of:
+    # # https://kubernetes.io/docs/concepts/storage/dynamic-provisioning
+    # # TODO: Remove PVC if Tekton version > = 0.12, use 'volumeClaimTemplate' instead
```

Comment:
can we remove this? we are on Tekton 0.13 + now

Reply:
We probably need to keep this comment because we are not yet using volumeClaimTemplate. We can rephrase the TODO comment to `TODO: Remove PVC and use 'volumeClaimTemplate' instead`

Reply:
Addressed